### PR TITLE
Added to demos metadata where demos are based on papers

### PR DESCRIPTION
**Pull request by BM7878**

### CODE REVIEW

**Comment by josh146** [comment]
Thanks @BM7878! I unfortunately won't have time to double check the DOI links on my end.

I think also https://pennylane.ai/qml/demos/tutorial_expressivity_fourier_series, https://pennylane.ai/qml/demos/tutorial_vqls/, and https://pennylane.ai/qml/demos/tutorial_implicit_diff_susceptibility should be included here?

**Comment by josh146** [comment]
@alvaro-at-xanadu will you have the time to review this one?

**Comment by BM7878** [comment]
> Thanks @BM7878! I unfortunately won't have time to double check the DOI links on my end.
> 
> I think also https://pennylane.ai/qml/demos/tutorial_expressivity_fourier_series, https://pennylane.ai/qml/demos/tutorial_vqls/, and https://pennylane.ai/qml/demos/tutorial_implicit_diff_susceptibility should be included here?

Hi @josh146 , the first two of those seem to already have their basedOnPapers field set. With the third one, it's not clear to me which of the references the demo is based on?

**Comment by josh146** [comment]
You're correct @BM7878! I didn't realize, thanks!

> With the third one, it's not clear to me which of the references the demo is based on?

This demo was written as a demo for this paper, https://arxiv.org/abs/2211.13765, by the same authors :)

**Comment by BM7878** [comment]
> This demo was written as a demo for this paper, https://arxiv.org/abs/2211.13765, by the same authors :)

Added @josh146

File: demonstrations/tutorial_barren_plateaus.metadata.json [hunk 1]
```diff
@@ -49,7 +49,7 @@
             "url": ""
         }
     ],
-    "basedOnPapers": [],
+    "basedOnPapers": ["10.48550/arXiv.1406.2572"],
```
Comment (KetpuntoG) [inline demonstrations/tutorial_barren_plateaus.metadata.json:52]: The paper should be this one: https://arxiv.org/abs/1803.11173

File: demonstrations/tutorial_general_parshift.metadata.json [hunk 2]
```diff
@@ -67,7 +67,7 @@
             "url": "https://arxiv.org/abs/2108.01218"
         }
     ],
-    "basedOnPapers": [],
+    "basedOnPapers": ["10.48550/arXiv.2107.12390", "10.48550/arXiv.1812.06323", "10.48550/arXiv.2107.08131", "10.48550/arXiv.2108.01218"],
```
Comment (KetpuntoG) [inline demonstrations/tutorial_general_parshift.metadata.json:70]: I would put just the first one in this list

**Review by KetpuntoG — COMMENTED** [review]
Awesome! 🚀
Just two comments :)

**Review by KetpuntoG — APPROVED** [review]
Awesome, thanks!

**Review by alvaro-at-xanadu — APPROVED** [review]
Looks good to me!

**Pull request merged**